Backend/src/v1/testing.py

```diff
@@ -136,3 +142,21 @@ async def manual_test_sequence(
         raise e
 
     return response
+
+@router.post("/simulate_labels", status_code=200)
+def simulate_labels_route(dataset_name: str, fraction: float = 0.1):
+    """
+    Simulate labels for a dataset and return the modified DataFrame as JSON.
+    """
+    df = get_nodes_from_latest_version(dataset_name=dataset_name)
+    try:
+        # Simulate user input and get the updated DataFrame
+        updated_df = simulate_user_input(df, dataset_name=dataset_name, fraction=fraction)
+        # Convert the updated DataFrame to a list of Node objects
+        list_of_nodes = [Node(**one_node) for one_node in updated_df.to_dict("records")]
+        # Persist the updated nodes in the database
+        add_versioned_nodes(nodes=list_of_nodes, dataset_name=dataset_name)
```

Comment:
This creates a new df instead of updating the version.

I would suggest one of two solutions:
1. Use `label_one_node`
2. Create a new function which works in the same way but for bulk

Either way, this is something that does not currently work 100%. @Sinoeng is working on it, so I think it would be good to sync with him.